src/consensus/aft/raft.h

```diff
@@ -673,6 +677,48 @@ namespace aft
       }
     }
 
+    void recv_view_change_evidence(const uint8_t* data, size_t size)
+    {
+      ViewChangeEvidenceMsg r;
+      try
+      {
+        r = channels
+              ->template recv_authenticated_with_load<ViewChangeEvidenceMsg>(
+                data, size);
+      }
+      catch (const std::logic_error& err)
+      {
+        LOG_FAIL_FMT("Error in recv_view_change_evidence message");
+        LOG_DEBUG_FMT(
+          "Error in recv_view_change_evidence message: {}", err.what());
+        return;
+      }
+
+      auto node = nodes.find(r.from_node);
+      if (node == nodes.end())
+      {
+        // Ignore if we don't recognise the node.
+        LOG_FAIL_FMT(
+          "Recv nonce reveal to {} from {}: unknown node",
+          state->my_node_id,
+          r.from_node);
+        return;
+      }
+
+      if (r.from_node != state->requested_evidence_from)
+      {
+        // Ignore if we didn't request this evidence.
+        LOG_FAIL_FMT(
+          "we received evidence which we did not request, ignoring it {}",
```

Comment:
There are no formal guidelines (yet?) but we generally capitalize error messages, and tend to omit the pronoun.